### PR TITLE
chore(docs): Add spec for `listen` option

### PR DESCRIPTION
**Pull request by jszwedko**
Creating partially as a strawman to push this conversation forward.

In https://github.com/vectordotdev/vector/pull/8858#discussion_r695992243 we attempted to codify the option name to use for addresses on which Vector components listen on. We couldn't come to a consensus then, so the change was backed out. Here I reintroduce a proposal. I considered both `bind` and `listen` and decided in `listen` mostly due to the fact that it seems to match more with socket terminology where `bind` just allocates the address and `listen` actually starts passively listening for clients (and then `accept` accepts a new connection).

Some prior art from other tools to consider:

* FluentD: `bind` and `port`
* Nginx: `listen`
* Apache HTTPD: `listen`
* OpenTelemetry Collector: `endpoint` for both binding and transmitting
* Fluent-Bit: `listen`
* Logstash: `host` and `port`
* Filebeat: `listen_address` and `listen_port`
* Telegraf: `service_address`

Closes: https://github.com/vectordotdev/vector/issues/8877

### CODE REVIEW

File: docs/specs/component.md [hunk 1]
```diff
@@ -84,6 +84,22 @@ representing multiple endpoints. If a component uses multiple options to
 automatically build the endpoint, then the `endpoint(s)` option MUST
 override that process.
 
+#### `listen`
+
+When a component listens for incoming connections, it SHOULD expose a `listen` option that takes
```
Comment (neuronull) [inline docs/specs/component.md:89]:
```suggestion
When a component listens for incoming connections, it SHOULD expose a `listen` configuration option that takes
```

**Review by bruceg — APPROVED** [review]
Based on your survey, I would agree that `listen` does have the most mind share among configurations.

File: docs/specs/component.md [hunk 2]
```diff
@@ -84,6 +84,22 @@ representing multiple endpoints. If a component uses multiple options to
 automatically build the endpoint, then the `endpoint(s)` option MUST
 override that process.
 
+#### `listen`
+
+When a component listens for incoming connections, it SHOULD expose a `listen` configuration option that takes
+a `string` representing an address with `<protocol>:<address>`.
```
Comment (spencergilbert) [inline docs/specs/component.md:90]: This is the user facing type, right? Can we clarify that we should have an internal type to represent these? (I'm assuming we'd want a type for this)

Comment (jszwedko) [reply docs/specs/component.md:90]: 🤔 I think the internal implementation is outside of the scope of this spec which just focuses on the UX for users.

Comment (spencergilbert) [reply docs/specs/component.md:90]: Fair enough.

**Review by neuronull — APPROVED** [review]

**Pull request merged**